executor/hash_table.go

```diff
@@ -237,6 +239,30 @@ func (c *hashRowContainer) PutChunk(chk *chunk.Chunk) error {
 	}
 	return nil
 }
+func (c *hashRowContainer) PutChunkSelected(chk *chunk.Chunk, selected []bool) error {
+	chkIdx := uint32(c.records.NumChunks())
+	numRows := chk.NumRows()
+
+	c.records.Add(chk)
+	c.hCtx.initHash(numRows)
+
+	hCtx := c.hCtx
+	for _, colIdx := range c.hCtx.keyColIdx {
+		err := codec.HashChunkColumns(c.sc, hCtx.hashVals, chk, hCtx.allTypes[colIdx], colIdx, hCtx.buf, hCtx.hasNull)
```

Comment:
OK, I implemente `codec.HashChunkSelectedColumns`
In fact as regard to performance, using the selected may introduce more branch mispredictions, so that it can degrade the performance.

Reply:
It is already existed. https://godoc.org/github.com/pingcap/tidb/util/codec#HashChunkSelected

Reply:
ok